File: FloatItWatch/src/main/java/net/geekstools/floatshort/PRO/Configurations.java
/*
 * Copyright © 2020 By Geeks Empire.
 *
 * Created by Elias Fazel
 * Last modified 4/25/20 12:13 PM
 *
 * Licensed Under MIT License.
 * https://opensource.org/licenses/MIT
 */

package net.geekstools.floatshort.PRO;

import android.Manifest;
import android.content.Context;
import android.content.Intent;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.provider.Settings;
import android.support.wearable.activity.WearableActivity;
import android.util.TypedValue;

import com.google.firebase.FirebaseApp;

import net.geekstools.floatshort.PRO.Shortcuts.ApplicationsViewWatch;
import net.geekstools.floatshort.PRO.Utils.Functions.FunctionsClass;
import net.geekstools.floatshort.PRO.Utils.Functions.PublicVariable;

public class Configurations extends WearableActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        FirebaseApp.initializeApp(getApplicationContext());

        FunctionsClass functionsClass = new FunctionsClass(getApplicationContext());
        functionsClass.loadSavedColor();

        if (!BuildConfig.DEBUG) {
            functionsClass.savePreference(".UserInformation", "isBetaTester", functionsClass.appVersionName(getPackageName()).equals("[BETA]") ? true : false);
            functionsClass.savePreference(".UserInformation", "installedVersionCode", functionsClass.appVersionCode(getPackageName()));
            functionsClass.savePreference(".UserInformation", "installedVersionName", functionsClass.appVersionName(getPackageName()));
            functionsClass.savePreference(".UserInformation", "deviceModel", functionsClass.getDeviceName());
            functionsClass.savePreference(".UserInformation", "userRegion", functionsClass.getCountryIso());

            if (functionsClass.appVersionName(getPackageName()).equals("[BETA]")) {
                functionsClass.saveDefaultPreference("JoinedBetaProgrammer", true);
            }
        }

        SharedPreferences sharedPreferences = getSharedPreferences("theme", Context.MODE_PRIVATE);
        PublicVariable.alpha = 133;
        PublicVariable.opacity = 255;
        if (!sharedPreferences.getBoolean("hide", false)) {
            PublicVariable.hide = false;
        } else if (sharedPreferences.getBoolean("hide", false)) {
            PublicVariable.hide = true;
        }
        PublicVariable.size = 33;
        PublicVariable.HW = (int) TypedValue.applyDimension(TypedValue.COMPLEX_UNIT_DIP, PublicVariable.size, getResources().getDisplayMetrics());

        String[] Permissions = {
                Manifest.permission.WAKE_LOCK,
                Manifest.permission.INTERNET,
                Manifest.permission.VIBRATE
        };
        requestPermissions(Permissions, 666);

        if (!Settings.canDrawOverlays(getApplicationContext())) {

            startActivity(new Intent(getApplicationContext(), PermissionDialogue.class));

        } else {
            functionsClass.updateRecoverShortcuts();

            Intent applicationsViewWatch = new Intent(getApplicationContext(), ApplicationsViewWatch.class);
            applicationsViewWatch.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
            startActivity(applicationsViewWatch);
        }


        finish();
    }

    @Override
    public void onEnterAmbient(Bundle ambientDetails) {
        super.onEnterAmbient(ambientDetails);
    }

    @Override
    public void onUpdateAmbient() {
        super.onUpdateAmbient();
    }

    @Override
    public void onExitAmbient() {
        super.onExitAmbient();
    }
}

File: FloatItWatch/src/main/java/net/geekstools/floatshort/PRO/Configurations.kt
/*
 * Copyright © 2020 By Geeks Empire.
 *
 * Created by Elias Fazel
 * Last modified 4/25/20 12:20 PM
 *
 * Licensed Under MIT License.
 * https://opensource.org/licenses/MIT
 */
package net.geekstools.floatshort.PRO

import android.Manifest
import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.provider.Settings
import android.support.wearable.activity.WearableActivity
import android.util.TypedValue
import com.google.firebase.FirebaseApp
import net.geekstools.floatshort.PRO.Shortcuts.ApplicationsViewWatch
import net.geekstools.floatshort.PRO.Utils.Functions.FunctionsClass
import net.geekstools.floatshort.PRO.Utils.Functions.PublicVariable

class Configurations : WearableActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        FirebaseApp.initializeApp(applicationContext)

        val functionsClass = FunctionsClass(applicationContext)
        functionsClass.loadSavedColor()

        if (!BuildConfig.DEBUG) {
            functionsClass.savePreference(".UserInformation", "isBetaTester", if (functionsClass.appVersionName(packageName) == "[BETA]") true else false)
            functionsClass.savePreference(".UserInformation", "installedVersionCode", functionsClass.appVersionCode(packageName))
            functionsClass.savePreference(".UserInformation", "installedVersionName", functionsClass.appVersionName(packageName))
            functionsClass.savePreference(".UserInformation", "deviceModel", functionsClass.deviceName)
            functionsClass.savePreference(".UserInformation", "userRegion", functionsClass.countryIso)

            if (functionsClass.appVersionName(packageName) == "[BETA]") {
                functionsClass.saveDefaultPreference("JoinedBetaProgrammer", true)
            }
        }


        PublicVariable.size = 33
        PublicVariable.HW = TypedValue.applyDimension(TypedValue.COMPLEX_UNIT_DIP, PublicVariable.size.toFloat(), resources.displayMetrics).toInt()

        val sharedPreferences = getSharedPreferences("theme", Context.MODE_PRIVATE)
        if (!sharedPreferences.getBoolean("hide", false)) {
            PublicVariable.hide = false
        } else if (sharedPreferences.getBoolean("hide", false)) {
            PublicVariable.hide = true
        }

        val allPermissions = arrayOf(
                Manifest.permission.WAKE_LOCK,
                Manifest.permission.INTERNET,
                Manifest.permission.VIBRATE
        )
        requestPermissions(allPermissions, 666)

        if (!Settings.canDrawOverlays(applicationContext)) {

            startActivity(Intent(applicationContext, PermissionDialogue::class.java))

        } else {

            functionsClass.updateRecoverShortcuts()

            val applicationsViewWatch = Intent(applicationContext, ApplicationsViewWatch::class.java)
            applicationsViewWatch.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK)
            startActivity(applicationsViewWatch)
        }

        finish()
    }
}